### PR TITLE
Admin status info

### PR DESCRIPTION
**Pull request by ccmoloney**
Added new formats to status page : QUEUE & ADMIN

QUEUE format (example URL: https://32dweg5wqd.execute-api.ap-southeast-2.amazonaws.com/LATEST/wps/jobStatus?format=queue) displays the jobs that are queued, running and completed.  It gives a quick overview of what is in the AWS Batch queues at the point in time it is run.  It is intended to be a tool for support folk to be able to find jobs that have been queued/processed recently.  

ADMIN format (example URL: https://wps2.dev.aodn.org.au/wps/jobStatus?jobId=50d55315-6a28-4188-b3a0-d8adf77d981e&format=admin) displays the basic status information (same as the HTML format) plus it outputs the request XML - so a support person can see what was submitted.

The queue view HTML is generated by running the job_queue_html_template.ftl freemarker template.

### CODE REVIEW

**Comment by jonescc** [comment]
Might be nice for the job id on the queue page to link to the admin status page for the job.

File: job-status-service/src/main/java/au/org/aodn/aws/wps/lambda/JobStatusServiceRequestHandler.java [hunk 1]
```diff
@@ -333,4 +388,131 @@ private String getStatusDescription(StatusType currentStatus) {
...
+    private String getRequestSummary(S3Object requestFileObject) {
+
+        //  Currently just returns the content of the request file.
+        //  Can be expanded to provide a richer HTML summary of the request for
+        //  admin purposes (possible including links to logs etc).
+        try {
+
+            if (requestFileObject != null) {
+
+                S3ObjectInputStream requestInputStream = requestFileObject.getObjectContent();
```
Comment (jonescc) [inline job-status-service/src/main/java/au/org/aodn/aws/wps/lambda/JobStatusServiceRequestHandler.java]: Need to ensure this is closed (try with resources would do the trick!)

File: job-status-service/src/main/java/au/org/aodn/aws/wps/lambda/JobStatusServiceRequestHandler.java [hunk 2]
```diff
@@ -333,4 +388,131 @@ private String getStatusDescription(StatusType currentStatus) {
...
+
+            ArrayList<String> jobIds = new ArrayList<>();
+            for (JobSummary summary : waitingJobSummaries) {
+                jobIds.add(summary.getJobId());
+            }
+
+            waitingJobDetails = AWSBatchUtil.getJobDetails(batchClient, jobIds);
+        }
+
+        List<JobDetail> runningJobDetails = null;
```
Comment (jonescc) [inline job-status-service/src/main/java/au/org/aodn/aws/wps/lambda/JobStatusServiceRequestHandler.java]: Basically a repeat of above with different objects - would it be better to reuse a single function to reduce the risk of this code dievreging/needing to make the same change in several places.

File: job-status-service/src/main/java/au/org/aodn/aws/wps/lambda/JobStatusServiceRequestHandler.java [hunk 3]
```diff
@@ -333,4 +388,131 @@ private String getStatusDescription(StatusType currentStatus) {
...
+            ArrayList<String> jobIds = new ArrayList<>();
+            for (JobSummary summary : runningJobSummaries) {
+                jobIds.add(summary.getJobId());
+            }
+
+            runningJobDetails = AWSBatchUtil.getJobDetails(batchClient, jobIds);
+        }
+
+
+        List<JobDetail> completedJobDetails = null;
```
Comment (jonescc) [inline job-status-service/src/main/java/au/org/aodn/aws/wps/lambda/JobStatusServiceRequestHandler.java]: Same here.

File: job-status-service/src/main/java/au/org/aodn/aws/wps/lambda/JobStatusServiceRequestHandler.java [hunk 4]
```diff
@@ -333,4 +388,131 @@ private String getStatusDescription(StatusType currentStatus) {
...
+
+            completedJobDetails = AWSBatchUtil.getJobDetails(batchClient, jobIds);
+        }
+
+
+        //  Invoke freemarker template
+        String templateS3Bucket = WpsConfig.getConfig(QUEUE_VIEW_HTML_TEMPLATE_S3_BUCKET_CONFIG_KEY);
+        String templateS3Key = WpsConfig.getConfig(QUEUE_VIEW_HTML_TEMPLATE_S3_KEY_CONFIG_KEY);
+
+        S3ObjectInputStream contentStream = S3Utils.getS3ObjectStream(templateS3Bucket, templateS3Key);
```
Comment (jonescc) [inline job-status-service/src/main/java/au/org/aodn/aws/wps/lambda/JobStatusServiceRequestHandler.java]: Need to ensure this is closed.

**Comment by ccmoloney** [comment]
Fixed the naughty code things (closing streams + repeated code).
Will add features like links to status pages etc next iteration once i get feedback from Jac, Kate + Seb (and anyone else who has ideas).

**Pull request merged**